package ssp;
message MsgHeader{
    optional uint32 ver = 1;
    optional uint32 cmd_type = 2;
};
message RegReq{
    optional MsgHeader header = 1; //cmd = 401
    optional string user_name = 3;
    optional string password = 4;
    optional uint32 from = 5;
};
message RegRsp{
    optional MsgHeader header = 1; //cmd = 402
    optional string user_name = 2;
    optional uint32 ret = 3;
    optional uint32 user_id = 4;
    
};

message LoginReq{
    optional MsgHeader header = 1; //cmd = 101
    optional string user_name = 3;
    optional string password = 4;
};

message LoginRsp{
    optional MsgHeader header = 1; //cmd = 102
    optional uint32 ret = 3;
    optional string user_name = 4;
    optional uint32 user_id = 5;
}
message LogoutReq{
    optional string user_id = 3;
};

message LogoutRsp{
    optional uint32 ret = 2;
}
message AddFriendReq{
    optional MsgHeader header = 1; 
    optional uint32 user_id = 2;
    optional uint32 other_id = 3;
};
message AddFriendRsp{
    optional MsgHeader header = 1; 
    optional uint32 ret = 2;
};
message DelFriendReq{
    optional MsgHeader header = 1; 
    optional uint32 user_id = 2;
    optional uint32 other_id = 3;
};
message DelFriendRsp{
    optional MsgHeader header = 1; 
    optional uint32 ret = 2;
};

message PublishMessageReq{
    optional MsgHeader header = 1; 
    optional uint32 user_id = 2;
    optional string content = 3;
};

message PublishMessageRsp{
    optional MsgHeader header = 1; 
    optional uint32 ret = 2;
};

message GetMessageListReq{
    optional MsgHeader header = 1; 
    optional uint32 user_id = 2;
};
//////////////////////////////////
message MessageItem{
    optional uint32 publisher_id = 1;
    optional uint32 publish_time = 2;
    optional  string content = 3;
};
message GetMessageListRsp{
    optional MsgHeader header = 1; 
    optional uint32 ret = 2;
    repeated MessageItem message_list = 3;//可重复结构，配合size使用
};
//////////////////////////////////
message GetPhotoReq{
    optional MsgHeader header = 1; 
    optional uint32 user_id = 2;
};
message GetPhotoRsp{
    optional MsgHeader header = 1; 
    optional uint32 ret = 2;
    optional uint32 last_publisher_id = 3;
    optional string last_publish_name = 4;
} 

message CommonReq{
    optional MsgHeader header = 1;
    optional RegReq reg_req = 2; 
    optional LoginReq login_req = 3; 
    optional LogoutReq logout_req = 4; 
    optional AddFriendReq add_friend_req = 5; 
    optional DelFriendReq del_friend_req = 6; 
    optional GetPhotoReq get_photo_req = 7; 
    optional GetMessageListReq get_message_list_req = 8;
    optional PublishMessageReq publish_message_req = 9;
};
message CommonRsp{
    optional MsgHeader header = 1;
    optional RegRsp reg_rsp = 2; 
    optional LoginRsp login_rsp = 3; 
    optional LogoutRsp logout_rsp = 4; 
    optional AddFriendRsp add_friend_rsp = 5; 

    optional DelFriendRsp del_friend_rsp= 6; 
    optional GetPhotoRsp get_photo_rsp = 7; 
    optional GetMessageListRsp get_message_list_rsp= 8;
    optional PublishMessageRsp publish_message_rsp = 9;
};
